src/main/kotlin/com/sourcegraph/cody/edit/actions/lenses/LensEditAction.kt

```diff
@@ -0,0 +1,68 @@
+package com.sourcegraph.cody.edit.actions.lenses
+
+import com.intellij.openapi.actionSystem.ActionUpdateThread
+import com.intellij.openapi.actionSystem.AnAction
+import com.intellij.openapi.actionSystem.AnActionEvent
+import com.intellij.openapi.actionSystem.DataKey
+import com.intellij.openapi.actionSystem.PlatformDataKeys
+import com.intellij.openapi.diagnostic.Logger
+import com.intellij.openapi.editor.Editor
+import com.intellij.openapi.project.DumbAware
+import com.intellij.openapi.project.Project
+import com.sourcegraph.cody.agent.protocol.TaskIdParam
+import com.sourcegraph.cody.config.CodyAuthenticationManager
+import com.sourcegraph.common.CodyBundle
+
+abstract class LensEditAction(
+    val editAction: (Project, AnActionEvent, Editor, TaskIdParam) -> Unit
+) : AnAction(), DumbAware {
+  private val logger = Logger.getInstance(LensEditAction::class.java)
+
+  fun getActionUpdateThread(): ActionUpdateThread {
+    return ActionUpdateThread.EDT
+  }
+
+  override fun update(event: AnActionEvent) {
+    val project = event.project ?: return
+    val hasActiveAccount = CodyAuthenticationManager.getInstance(project).hasActiveAccount()
+    event.presentation.isEnabled = hasActiveAccount
+    if (!event.presentation.isEnabled) {
+      event.presentation.description =
+          CodyBundle.getString("action.sourcegraph.disabled.description")
+    }
+  }
+
+  override fun actionPerformed(e: AnActionEvent) {
+    try {
+      var project = e.project
+      if (project == null) {
+        project = e.dataContext.getData(PlatformDataKeys.PROJECT.name) as? Project
+      }
+      if (project == null || project.isDisposed) {
+        logger.warn("Received code lens action for null or disposed project: $e")
```

Comment:
This probably should be an error.

Reply:
Not really, it can happen if user do an edit and close project at the same time